src/Compilers/VisualBasic/Test/Semantic/DeclaringSyntaxNodeTests.vb

```diff
@@ -640,6 +640,42 @@ End Class
             CheckDeclaringSyntaxIsNoDeclaration(Of ForEachStatementSyntax)(comp, tree, "loc10")
         End Sub
 
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/6209")>
+        Public Sub CatchVariableDeclaringSyntax()
+            Dim comp = CompilationUtils.CreateCompilationWithMscorlib40(
+    <compilation name="CatchVariableDeclaringSyntax">
+        <file name="a.vb">
+Imports System
+
+Class C1
+    Sub m()
+        Try
+            Console.WriteLine()
+        Catch exc As Exception
+            Console.WriteLine(exc.ToString())
+        End Try
+    End Sub
+End Class
+    </file>
+    </compilation>)
+
+            Dim tree = comp.SyntaxTrees(0)
+            Dim model = comp.GetSemanticModel(tree)
+            Dim code As String = tree.GetText().ToString()
+            Dim position As Integer = code.IndexOf("exc", StringComparison.Ordinal)
+            Dim token = tree.GetCompilationUnitRoot().FindToken(position)
+            Dim identifierName = token.Parent
```

Comment:
>Dim identifierName = token.Parent [](http://example.com/codeflow?start=12&length=33)

This calculation looks unnecessary complicated. We simply could traverse the syntax tree for the first IdentifierNameSyntax with the right identifier value.